pyscf/adc/uadc_ao2mo.py

```diff
@@ -83,3 +89,239 @@ def transform_integrals_incore(myadc):
+            vvvv -= np.ascontiguousarray(vvvv.transpose(0,1,3,2))
+            vvvv = vvvv[:, :, ind_vv_g[0], ind_vv_g[1]]
+
+            vvvv_p = write_dataset(vvvv)
+            del vvvv
+            eris.vvvv_p.append(vvvv_p)       
+
+
+        for p in range(0,vir_b.shape[1],chnk_size):
+
+            if chnk_size < vir_b.shape[1] :
+                orb_slice = vir_b[:, p:p+chnk_size]
+            else :
+                orb_slice = vir_b[:, p:]
+
+            _, tmp = tempfile.mkstemp()
+            ao2mo.outcore.general(mol, (orb_slice, vir_b, vir_b, vir_b), tmp, max_memory = avail_mem, ioblk_size=100, compact=False)
+            VVVV = read_dataset(tmp,'eri_mo')
+            del (tmp)
+            VVVV = VVVV.reshape(orb_slice.shape[1], vir_b.shape[1], vir_b.shape[1], vir_b.shape[1])
+            VVVV = np.ascontiguousarray(VVVV.transpose(0,2,1,3))
+            VVVV -= np.ascontiguousarray(VVVV.transpose(0,1,3,2))
+            VVVV = VVVV[:, :, ind_VV_g[0], ind_VV_g[1]]
+
+            VVVV_p = write_dataset(VVVV)
+            del VVVV
+            eris.VVVV_p.append(VVVV_p)       
+
+
+        for p in range(0,vir_a.shape[1],chnk_size):
+
+            if chnk_size < vir_a.shape[1] :
+                orb_slice = vir_a[:, p:p+chnk_size]
+            else :
+                orb_slice = vir_a[:, p:]
+
+            _, tmp = tempfile.mkstemp()
+            ao2mo.outcore.general(mol, (orb_slice, vir_a, vir_b, vir_b), tmp, max_memory = avail_mem, ioblk_size=100, compact=False)
+            vVvV = read_dataset(tmp,'eri_mo')
+            del (tmp)
+            vVvV = vVvV.reshape(orb_slice.shape[1], vir_a.shape[1], vir_b.shape[1], vir_b.shape[1])
+            vVvV = np.ascontiguousarray(vVvV.transpose(0,2,1,3))
+            vVvV = vVvV.reshape(-1, vir_b.shape[1], vir_a.shape[1] * vir_b.shape[1])
+
+            vVvV_p = write_dataset(vVvV)
+            del vVvV
+            eris.vVvV_p.append(vVvV_p)       
+
+
+        for p in range(0,vir_b.shape[1],chnk_size):
+
+            if chnk_size < vir_b.shape[1] :
+                orb_slice = vir_b[:, p:p+chnk_size]
+            else :
+                orb_slice = vir_b[:, p:]
+
+            _, tmp = tempfile.mkstemp()
+            ao2mo.outcore.general(mol, (orb_slice, vir_b, vir_a, vir_a), tmp, max_memory = avail_mem, ioblk_size=100, compact=False)
+            VvVv = read_dataset(tmp,'eri_mo')
+            del tmp
+            VvVv = VvVv.reshape(orb_slice.shape[1], vir_b.shape[1], vir_a.shape[1], vir_a.shape[1])
+            VvVv = np.ascontiguousarray(VvVv.transpose(0,2,1,3))
+            VvVv = VvVv.reshape(-1, vir_a.shape[1], vir_b.shape[1] * vir_a.shape[1])
+
+            VvVv_p = write_dataset(VvVv)
+            del VvVv
+            eris.VvVv_p.append(VvVv_p)       
+    
+        cput2 = logger.timer_debug1(myadc, 'transforming vvvv', *cput2)
+
+    log.timer('ADC integral transformation', *cput0)
+    return eris
+
+
+def read_dataset(h5file, dataname):
```

Comment:
Are these shared by RADC and UADC? If so, UADC should use those from RADC.

Reply:
Done.

Reply:
Done. UADC using those from RADC.